sources/bloodhound/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+        .lines()
+        .any(|line| line.unwrap_or_default().contains(search_str)))
+}
+
+/// Reads a file and checks if the given `search_strs` are present in its contents.
+pub fn look_for_strings_in_file(path: &str, search_strs: &[&str]) -> Result<bool, io::Error> {
+    let mut matched = 0;
+
+    let reader = BufReader::new(File::open(path)?);
+    for line in reader.lines() {
+        let content = line.unwrap_or_default();
+        for search_str in search_strs {
+            if content.contains(search_str) {
+                matched += 1;
+            }
+        }
+    }
+
+    Ok(search_strs.len() <= matched)
+}
+
+/// Check if a given file contains all provided strings, getting a `CheckerResult` of its findings.
+///
+/// If all `strings_to_match` are found in `path`, the `CheckerResult` returned will have a `CheckStatus::PASS`.
+///
+/// If one or more of `strings_to_match` are not found in `path`, then even if some are found the `CheckerResult`
+/// returned will have a `CheckStatus::FAIL` and the `error` field will contain `unable_to_find_error` as its content.
+///
+/// If `path` cannot be read or there is some other error checking the content of the file, returned status will be
+/// `CheckStatus::SKIP` indicating a manual check will need to be performed and the `error` field will contain the
+/// `unable_to_check_error` value.
+#[macro_export]
+macro_rules! check_file_contains {
+    ($path:expr, $strings_to_match:expr, $unable_to_check_error:expr, $unable_to_find_error:expr) => {{
+        let mut result = results::CheckerResult {
+            error: String::new(),
+            status: results::CheckStatus::SKIP,
+        };
+
+        if let Ok(found) = look_for_strings_in_file($path, $strings_to_match) {
+            if found {
+                result.status = results::CheckStatus::PASS;
+            } else {
+                result.error = $unable_to_find_error.to_string();
+                result.status = results::CheckStatus::FAIL;
+            }
+        } else {
+            result.error = $unable_to_check_error.to_string();
+        }
+
+        result
+    }};
+}
+
+/// Executes a command and checks if the given `search_str` is in the output.
+pub fn look_for_string_in_output<I, S>(cmd: &str, args: I, search_str: &str) -> Option<bool>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    if let Ok(output) = Command::new(cmd).args(args).output() {
+        if output.status.success() {
+            let mut found = false;
+            let mp_output = String::from_utf8_lossy(&output.stdout).to_string();
+            for line in mp_output.lines() {
+                found |= line.contains(search_str);
+            }
+
+            Some(found)
+        } else {
+            None
+        }
+    } else {
+        None
+    }
+}
+
+/// Executes a command and checks if the given `search_strs` are in the output.
+pub fn look_for_strings_in_output<I, S>(cmd: &str, args: I, search_strs: &[&str]) -> Option<bool>
```

Comment:
It could - can you explain why we would want to avoid passing ownership of the args here though?

Reply:
If I consume this library, I can call this function with my parameters. But, in my use case I may want to use `args` after I call this function, but since I gave ownership of `args` I can't. What I would end up doing is `args.clone()` and pass that to keep my original `args` and don't lose ownership.

I guess TL;DR will be: don't take ownership if you don't absolutely need it.

Reply:
There are no cases I have seen where a consumer would need to still own those args. But unfortunately, due to `Command::args()`'s signature, it wouldn't do us much good anyway. We would need to copy those values over to a new Iterator. That would be worse that requiring a potential future library consumer to have to redeclare command arguments for a second use.